change/@ni-nimble-components-44c2b207-def9-472c-ae87-f6e48877185e.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Discussion topic, not blocking this PR: Would we ever want to add this feature to other inputs? Should we track that request? If it's low priority then I'm ok if we don't but just want to verify that it is low priority.
- text input: I've seen it used to clear search fields
- combobox: might be a natural change for "parity" with select (though users can achieve the same outcome with Delete)
- number: maybe for parity with text input but probably not a common use case
- text area and rich text editor: not likely useful

Reply:
I agree that we should add this to the combobox – clearing the Combobox with ESC isn't very discoverable today, and this will be more consistent and discoverable.

Filed https://github.com/ni/nimble/issues/2008

I also agree that this will be a requirement for a future search field, but that the other text areas don't need it.

Reply:
@fredvisser you'd suggested in a demo today that the SLE search field could have a clear button. I'm not sure how we'd recommend they do that today. Nimble doesn't yet offer a dedicated search field. The text input does include an [action slot](https://nimble.ni.dev/storybook/?path=/story/components-text-field--underline-text-field&args=label:Underline+Text+Field;type:text;appearance:underline;fullBleed:!false;value:;readonly:!false;disabled:!false;errorVisible:!false;errorText:Value+is+invalid;actionButton:!true;leftIcon:!false) which could contain a custom clear button, but it wouldn't exactly match the behavior we're proposing for the select (ESC would lose focus but not clear, action button would be keyboard focusable but clear button isn't).

Given this open question and the fact that the request is scope growth over sharing the existing search component sharing refactor, I'm not planning to drive this suggestion now with the implementing team. If you want to pick a direction track it as follow-up work, feel free.

Reply:
#2021